src/web3-adapter/customRPC.ts

```diff
@@ -0,0 +1,27 @@
+// web3-core-method exports both a function and a type called `Method`. Typescript defaults to importing the type rather than the function, so when I try to call it typescript will yell at me. So we use require instead.
```

Comment:
It sounds like you were writing
```ts
import { Method } from "web3-core-method";
```
because that's the only way to import a type, but that would be equivalent to
```ts
const { Method } = require("web3-core-method");
```
The line that's currently here in the PR would be equivalent instead to
```ts
import Method from "web3-core-method";
```
The difference is that `import Method` tries to take a "default import," or if none exists then the entire module, and assigns it to the variable `Method`, while `import { Method }` imports just the single export named `Method` from the module.

Reply:
Using `import Method from "web3-core-method";` will result in

```
This expression is not constructable.
  Type 'typeof import("/Users/benjamingodlove/Developer/alchemy-web3/node_modules/web3-core-method/types/index")' has no construct signatures.

18     const newMethod = new Method({
                             ~~~~~~
```
Basically it's just as confused because there are two exported `Method`s in the package. I've resolved this (in an upcoming commit) by allowing `Method` to be type `any`, and then at runtime it will choose the right implementation based on usage.

```
const MethodFn: any = Method;
```

Thanks for circling me back to this! It was bothering me quite a bit.

Reply:
Ah I see. The problem here is in the type definitions for the "web3-core-method" package. It's actually not because there's two exported `Method` names- when you write
```ts
import Method from "web3-core-method";
```
it doesn't actually have any connection to things named "Method" inside the package. That is, it does the same thing as if you wrote
```ts
import Blah from "web3-core-method";
```
just assigning the import to a different variable name. The issue is that the type definitions simply don't say that the default import is a function.

You see this happen a lot with packages that aren't written in TypeScript because the type definitions are likely written by a different person than whoever wrote the code (and especially for internal-ish packages like this one). Just have to live with the invalid definitions and cast to `any` as needed like you did here.

From a language perspective, TypeScript can't really get confused between a type and a value with the same name because when it's used, the grammar is unambiguous about whether it's a type or a value. The only place where it's ambiguous is the import statement itself, i.e.
```ts
import { Foo } from "bar";
```
but that's okay because at any usage site it will be definitively one or the other, so TypeScript can look ahead to see which type of import is needed.